### PR TITLE
Don't use a mutable map to record Methods

### PR DESCRIPTION
**Pull request by cquiroz**
As discussed on #956 we should try to refactor methods that use mutable structures or reflection. This PR removes the use of a mutable map to record

### CODE REVIEW

File: core/src/main/scala/org/http4s/Method.scala [hunk 1]
```diff
@@ -54,54 +53,55 @@ object Method extends MethodInstances {
...
+  val MKACTIVITY: Method = new Method("MKACTIVITY") with Idempotent
+  val MKCALENDAR: Method = new Method("MKCALENDAR") with Idempotent
+  val MKCOL: Method = new Method("MKCOL") with Idempotent
+  val MKREDIRECTREF: Method = new Method("MKREDIRECTREF") with Idempotent
+  val MKWORKSPACE: Method = new Method("MKWORKSPACE") with Idempotent
+  val MOVE: Method = new Method("MOVE") with Idempotent
+  val OPTIONS: Method = new Method("OPTIONS") with Safe with PermitsBody
+  val ORDERPATCH: Method = new Method("ORDERPATCH") with Idempotent
+  val PATCH: Method = new Method("PATCH") with Default with PermitsBody
+  val POST: Method = new Method("POST") with Default with PermitsBody
```
Comment (cquiroz) [inline core/src/main/scala/org/http4s/Method.scala]: On the list of methods there is a `PRI` method not present here. It seems to be an http2 method. Any reason it wasn't included here?

File: core/src/main/scala/org/http4s/Method.scala [hunk 2]
```diff
@@ -54,54 +53,97 @@ object Method extends MethodInstances {
...
+  val TRACE: SafeMethodWithBody = new Method("TRACE") with Safe with PermitsBody
+  val UNBIND: IdempotentMethod = new Method("UNBIND") with Idempotent
+  val UNCHECKOUT: IdempotentMethod = new Method("UNCHECKOUT") with Idempotent
+  val UNLINK: IdempotentMethod = new Method("UNLINK") with Idempotent
+  val UNLOCK: IdempotentMethod = new Method("UNLOCK") with Idempotent
+  val UPDATE: IdempotentMethod = new Method("UPDATE") with Idempotent
+  val UPDATEREDIRECTREF: IdempotentMethod = new Method("UPDATEREDIRECTREF") with Idempotent
+  val `VERSION-CONTROL`: IdempotentMethod = new Method("VERSION-CONTROL") with Idempotent
+
+  val all = List(
```
Comment (rossabaker) [inline core/src/main/scala/org/http4s/Method.scala]: This is going to be easy to forget to update, but that happens basically never.

Comment (cquiroz) [reply core/src/main/scala/org/http4s/Method.scala]: Right, that's a limitation. We could do an automated tests that loads the spec on xml and verifies but it seems too much effort

Comment (rossabaker) [reply core/src/main/scala/org/http4s/Method.scala]: Another automated test could be done through reflection, but that test wouldn't run on scala.js, which is part of the point.

**Comment by rossabaker** [comment]
Hmm, we never heard from appveyor on this one.  I'm not worried about Windows bugs on this one, but, weird.

**Pull request merged**